core/org.openjdk.jmc.common/src/main/java/org/openjdk/jmc/common/unit/QuantityRange.java

```diff
@@ -242,6 +242,15 @@ public String displayUsing(String formatHint) {
 		return minEnd.compareTo(maxStart) >= 0 ? createWithEnd(maxStart, minEnd) : null;
 	}
 
+	public static IRange<IQuantity> union(IRange<IQuantity> a, IRange<IQuantity> b) {
+		IQuantity minStart = a.getStart().compareTo(b.getStart()) < 0 ? a.getStart() : b.getStart();
+		IQuantity maxEnd = a.getEnd().compareTo(b.getEnd()) < 0 ? b.getEnd() : a.getEnd();
+		if (intersection(a, b) == null) { // if disjoint, no union
+			return null;
+		}
```

Comment:
Could we do this without potentially allocating a new QuantityRange object here? I.e. if the ranges intersect we'll allocate a new IRange just for a null check.